src/packages/block/block-rte/context/block-rte-manager.context.ts

```diff
@@ -0,0 +1,54 @@
+import type { UmbBlockRteLayoutModel, UmbBlockRteTypeModel } from '../types.js';
+import type { UmbBlockRteWorkspaceData } from '../index.js';
+import type { UmbBlockDataType } from '../../block/types.js';
+import type { Editor } from '@umbraco-cms/backoffice/external/tinymce';
+import { UmbBlockManagerContext } from '@umbraco-cms/backoffice/block';
+
+import '../components/block-rte-entry/index.js';
+
+/**
+ * A implementation of the Block Manager specifically for the Rich Text Editor.
+ */
+export class UmbBlockRteManagerContext<
+	BlockLayoutType extends UmbBlockRteLayoutModel = UmbBlockRteLayoutModel,
+> extends UmbBlockManagerContext<UmbBlockRteTypeModel, BlockLayoutType> {
+	//
+	#editor?: Editor;
+
+	setTinyMceEditor(editor: Editor) {
+		this.#editor = editor;
+	}
+
+	create(
+		contentElementTypeKey: string,
+		partialLayoutEntry?: Omit<BlockLayoutType, 'contentUdi'>,
+		modalData?: UmbBlockRteWorkspaceData,
+	) {
+		return super.createBlockData(contentElementTypeKey, partialLayoutEntry);
+	}
+
+	insert(
+		layoutEntry: BlockLayoutType,
+		content: UmbBlockDataType,
+		settings: UmbBlockDataType | undefined,
+		modalData: UmbBlockRteWorkspaceData,
+	) {
+		if (!this.#editor) return false;
+
+		if (layoutEntry.displayInline) {
```

Comment:
This never seems to be "true" even though I set it up for my block, so the block is always inserted as a "display:block" element:

![image](https://github.com/umbraco/Umbraco.CMS.Backoffice/assets/752371/01796d8d-b461-49b3-80ad-462807d0ac45)

Also, I noticed something for custom_elements in the TinyMCE documentation, where they support specifying [custom_elements](https://www.tiny.cloud/docs/tinymce/6/content-filtering/#custom_elements) (and have been for a while), so we might consider using that instead of our custom pre-init filter to unwrap inline elements:

```js
custom_elements: 'umb-rte-block,~umb-rte-block-inline',
```

Reply:
Agree on the content editable. You should be able to do that using the pre-init filter.

Reply:
But for inline, I didn't see it as missing CSS. Because the property was not true, it didn't even seem to insert the inline element.

Reply:
But I fixed the UI/Action-bar and the Icons, which needed another interesting Event-Proxy. So now its far more complete. :-)